### PR TITLE
[7.x] Adds the `redacted` string helper methods

### PR DESCRIPTION
**Pull request by DarkGhostHunter**
## Redact (obfuscate) part of a string

This pull includes three new helpers that allow developers to redact (obfuscate) part of a string. This can be used to hide parts of strings like credit card numbers, emails, or several iterations of a given value.

It redacts all unicode word characters and digits.

### `redact`

This basically redacts half of the string by default. You can set a custom length.

```php
echo Str::redact('5467 3665 1540 5769'); // **** **** 1540 5769
echo Str::redact('5467 3665 1540 5769', 14); // **** **** **** 5769
```

### `redactAll`

This redacts all word iterations inside a string for another string.

```php
echo Str::redactAll('Michael called to another Michael'); // ******* called to another *******
```

### `redactBefore`

This redacts the string until a given value.

```php
echo Str::redactBefore('jan.michael.vincent@gmail.com', '@'); *******************@gmail.com
```

> This sweeps all the characters instead of just word characters.

## Todo

* Maybe create `redactWords` to just redact words until certain number of words. I will literally need help on that.

### CODE REVIEW

**Comment by DarkGhostHunter** [comment]
F*ck it, I went and re-worked it. Updated first post.

**Comment by netpok** [comment]
You're test are failing and based on the output there is a dump left in it somewhere. Also I think it would be great to support negative length: `redact -4 => keep the last 4 only`.

File: tests/Support/SupportStrTest.php [hunk 1]
```diff
@@ -192,8 +192,49 @@ public function testStrContainsAll()
...
+
+        $this->assertSame('@@@ bar @@@', Str::redactAll('foo bar foo', 'foo', '@'));
+        $this->assertSame('@是一段中@', Str::redactAll('这是一段中这', '这', '@'));
+        $this->assertSame('foo bar foo', Str::redactAll('foo bar foo', 'quz', '123'));
+        $this->assertSame('这是一段中这', Str::redactAll('这是一段中这', 'ミ', '123'));
+    }
+
     public function testParseCallback()
     {
+        echo Str::redact('Jan Michael Vincent');
```
Comment (netpok) [inline tests/Support/SupportStrTest.php:236]: Actually it was an echo

**Comment by GrahamCampbell** [comment]
Personally, I don't think this belongs in the framework core, tbh.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**